src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+            ? elem.ownerDocument
+            : document
+          : null;
+
+        if (!ownerDocument) {
+          return;
+        }
+        requestAnimationFrame(() => {
+          const focusInPopover = this.popoverListRef.current?.contains(
+            ownerDocument.activeElement,
+          );
+          const focusInInput =
+            ownerDocument.activeElement === this.filterInputRef.current;
+          const focusInCombobox = focusInPopover || focusInInput;
+          setPopoverVisibility(focusInCombobox);
+
+          if (!focusInCombobox) {
+            this.setState(
+              (
+                _: ComboboxState<T & ComboboxData>,
+                prevProps: ComboboxProps<T & ComboboxData>,
+              ) => ({
+                filterInputValue: '',
+                filteredItems: prevProps.items,
+              }),
+            );
+          }
+        });
+      }
+    };
+
+    const setPopoverVisibility = (toState: Boolean) => {
+      this.setState({ showPopover: toState });
+    };
+
+    const {
+      filteredItems,
+      showPopover,
+      currentSelection,
+      selectedKeys,
+      selectedItems,
+      disabledKeys,
+      filterInputValue,
+    } = this.state;
+
+    const focusToMenu = () => {
+      if (
+        this.popoverListRef !== null &&
+        this.popoverListRef.current !== null &&
+        showPopover
+      ) {
+        this.popoverListRef.current.focus();
+      }
+    };
+
+    const {
+      id,
+      className,
+      items: propItems,
+      labelText,
+      onItemSelectionsChange,
+      chipListVisible,
+      chipActionLabel,
+      removeAllButtonLabel,
+      visualPlaceholder,
+      emptyItemsLabel,
+      defaultSelectedItems,
+      onChange: propOnChange,
+      debounce,
+      status,
+      statusText,
+      selectedItems: controlledItems,
+      onItemSelect,
+      onRemoveAll,
+      ...passProps
+    } = this.props;
+
+    const selectedAndDisabledKeys = Object.keys(selectedKeys).reduce(
+      (amount: number, key: string) => {
+        if (key in disabledKeys) {
+          return amount + 1;
+        }
+        return amount;
+      },
+      0,
+    );
+
+    const showChipList =
+      chipListVisible && selectedItems && selectedItems.length > 0;
+
+    const showRemoveAllButton =
+      removeAllButtonLabel &&
+      Object.keys(selectedKeys).length > 0 &&
+      Object.keys(selectedKeys).length !== selectedAndDisabledKeys;
+
+    return (
+      <HtmlDiv
+        role="combobox"
+        aria-haspopup="listbox"
+        aria-controls={`${id}-popover`}
+        aria-expanded={showPopover}
+        {...passProps}
+        className={classnames(baseClassName, className, {
+          [comboboxClassNames.open]: showPopover,
+          [comboboxClassNames.error]: status === 'error',
+        })}
+      >
+        <HtmlDiv className={classnames(comboboxClassNames.wrapper, {})}>
+          <Debounce waitFor={debounce}>
+            {(debouncer: Function) => (
```

Comment:
Is this the convention we have been using before? Naming sounds funny, but it's understandable so I guess it's good as it is.

Reply:
`TextInput`, `SearchInput` and `Debounce` uses this naming at this time so was following the naming.

Is there suggestion for more describing name?

Reply:
I'd prefer something like `debounceFn` etc. But if this is used elsewhere I propose we keep using the same convention.